Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -96,31 +96,108 @@ final class CacheAdvanceTests: XCTestCase {
         XCTAssertEqual(messages, [])
     }
 
-    func test_messages_throwsFileCorruptedWhenOffsetInFileAtEndOfNewsetMessageOutOfSync() throws {
-        let randomHighValue: UInt64 = 10_1000
+    func test_messages_throwsFileCorruptedWhenOffsetInFileAtEndOfNewestMessageIsBeyondEndOfNewestMessageButBeforeEndOfFile() throws {
+        let message: TestableMessage = "This is a test"
+        let requiredByteCount = try self.requiredByteCount(for: [message])
+        let maximumBytes = requiredByteCount + 2
         let header = try CacheHeaderHandle(
             forReadingFrom: testFileLocation,
-            maximumBytes: randomHighValue,
+            maximumBytes: maximumBytes,
             overwritesOldMessages: true)
-        let cache = CacheAdvance<TestableMessage>(
-            fileURL: testFileLocation,
-            writer: try FileHandle(forWritingTo: testFileLocation),
-            reader: try CacheReader(forReadingFrom: testFileLocation),
-            header: try CacheHeaderHandle(
-                forReadingFrom: testFileLocation,
-                maximumBytes: header.maximumBytes,
-                overwritesOldMessages: header.overwritesOldMessages),
-            decoder: JSONDecoder(),
-            encoder: JSONEncoder())
-
-        // Make sure the header data is persisted before we read it as part of the `messages()` call below.
-        try header.synchronizeHeaderData()
-        // Our file is empty. Make the file corrupted by setting the offset at end of newest message to be further in the file.
-        // This should never happen, but past versions of this repo could lead to a file having this kind of inconsistency if a crash occurred at the wrong time.
+
+        func makeCache() throws -> CacheAdvance<TestableMessage> {
+            return CacheAdvance<TestableMessage>(
+                fileURL: testFileLocation,
+                writer: try FileHandle(forWritingTo: testFileLocation),
+                reader: try CacheReader(
+                    forReadingFrom: testFileLocation,
+                    maximumBytes: maximumBytes),
+                header: header,
+                decoder: JSONDecoder(),
+                encoder: JSONEncoder())
+        }
+        let writingCache = try makeCache()
+        try writingCache.append(message: message)
+
+        // Make the file corrupted by setting the offset at end of newest message to be further in the file.
+        // This could happen if a crash occurred during a write of `header.offsetInFileAtEndOfNewestMessage` on a big-endian device.
+        try header.updateOffsetInFileAtEndOfNewestMessage(
+            to: requiredByteCount + 1)
+
+        // Create a new cache instance that uses the corrupted data persisted to disk
+        let corruptedReadingCache = try makeCache()
+
+        XCTAssertThrowsError(try corruptedReadingCache.messages()) {
```

Comment:
I want to make sure I follow: what was the behavior before this test?

Reply:
I should have left a comment! The behavior of this test before the fix was an infinite loop – this test would not complete.